### PR TITLE
WINDUPRULE-73: Refactoring of environment-dependent-calls

### PR DESCRIPTION
**Pull request by mbriskar**

### CODE REVIEW

File: rules-reviewed/jboss-eap/uncategorized/environment-dependent.windup.xml [hunk 1]
```diff
@@ -64,19 +63,37 @@
...
+                    <hint title="Proprietary InitialContext initialization" effort="1" severity="mandatory">
+                    <message>In JBoss EAP, the InitialContext should be instantiated with no arguments. Once an instance is constructed, look up the service using portable JNDI lookup syntax. Ensure also that in case system properties for InitialContext are provided, they do not need to be changed for the JBoss.
+
+                        `InitialContext context = new InitialContext();
+                        Service service = (Service) context.lookup( "java:app/service/" + ServiceImpl.class.getSimpleName() );`
+                        </message>
+                  <!-- <link href="https://access.redhat.com/articles/1496973" title="Migrate Oracle WebLogic Applications to Use Portable JNDI Syntax in Red Hat JBoss Enterprise Application Platform" /> -->
```
Comment (mbriskar) [inline rules-reviewed/jboss-eap/uncategorized/environment-dependent.windup.xml]: here is the link to weblogic specific migration article that is however quite generic

File: rules-reviewed/jboss-eap/uncategorized/environment-dependent.windup.xml [hunk 2]
```diff
@@ -24,7 +23,7 @@
                 </javaclass>
             </when>
             <perform>
-                 <hint title="Dynamic instantiation of class" effort="0" severity="optional">
+                 <hint title="Dynamic instantiation of class" effort="1" severity="mandatory">
```
Comment (mbriskar) [inline rules-reviewed/jboss-eap/uncategorized/environment-dependent.windup.xml]: I have added 1 extra story point for Class.forName() even though we didn't agree on it in the issue. I think it makes sense, or does it not?

File: rules-reviewed/jboss-eap/uncategorized/tests/data/MyClass.java [hunk 3]
```diff
@@ -12,7 +14,10 @@ public MyClass(){
    public void myMethod() {
 	Class.forName("environment.dependent.MyClass");
         Class.forName("environment.dependent.OtherClass");
-        InitialContext context = new InitialContext();
+        Properties environment = new Properties();
```
Comment (mbriskar) [inline rules-reviewed/jboss-eap/uncategorized/tests/data/MyClass.java]: I'd like to note that I am testing this with Properties that is a subclass of Hashtable and it works like a charm (I was not sure 100 percent sure if we support this).

Comment (jsight) [inline rules-reviewed/jboss-eap/uncategorized/tests/data/MyClass.java]: Yes, you did a great job on that type binding code :)

Comment (mbriskar) [inline rules-reviewed/jboss-eap/uncategorized/tests/data/MyClass.java]: I think we thank this to Eclipse JDT that does the linking of the method properly :)

File: rules-reviewed/jboss-eap/uncategorized/environment-dependent.windup.xml [hunk 4]
```diff
@@ -13,7 +13,6 @@
         </dependencies>
         <targetTechnology id="eap" versionRange="[6,)" />
         <targetTechnology id="java-ee" versionRange="[6,)" />
-        <phase>PostMigrationRulesPhase</phase>
```
Comment (mbriskar) [inline rules-reviewed/jboss-eap/uncategorized/environment-dependent.windup.xml]: I don't think these rules are catch-all. Are they?

**Comment by jsight** [comment]
This looks good.

**Pull request merged**